src/main/java/bossmonster/controller/BossMonsterGame.java

```diff
@@ -0,0 +1,129 @@
+package bossmonster.controller;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import bossmonster.domain.AttackType;
+import bossmonster.domain.BossMonster;
+import bossmonster.domain.Health;
+import bossmonster.domain.Player;
+import bossmonster.domain.RandomDamageGenerator;
+import bossmonster.view.InputView;
+import bossmonster.view.OutputView;
+
+public class BossMonsterGame {
+
+	private final RandomDamageGenerator randomDamageGenerator;
+
+	public BossMonsterGame(RandomDamageGenerator randomDamageGenerator) {
+		this.randomDamageGenerator = randomDamageGenerator;
+	}
+
+	public void startGame() {
+		BossMonster bossMonster = repeat(this::inputBossHp);
+		Player player = repeat(this::inputPlayer);
+		OutputView.printStartRaidMessage();
+		int count = 0;
+		OutputView.printStartGameStatus(bossMonster, player);
+		play(bossMonster, player, count);
+	}
+
+	private void play(BossMonster bossMonster, Player player, int count) {
+		while (true) {
+			count++;
+			OutputView.printAttackType();
+			AttackType type = repeat(() -> inputAttackType(player));
+			attackBossByPlayer(bossMonster, player, type);
+			int damage = attackPlayerByBoss(player);
+			OutputView.printPlayerAttackDamage(type);
+			if (checkBossHpZero(bossMonster, player, count)) {
+				break;
+			}
+			OutputView.printBossAttackDamage(damage);
+			if (checkPlayerHpZero(bossMonster, player)) {
+				break;
+			}
+			OutputView.printPlayGameStatus(bossMonster, player);
+		}
+	}
+
+	private boolean checkPlayerHpZero(BossMonster bossMonster, Player player) {
+		if (player.playerHpZero()) {
+			OutputView.printFailGameStatus(bossMonster, player);
+			OutputView.printPlayerHpZero(player.getName());
+			return true;
+		}
+		return false;
+	}
+
+	private boolean checkBossHpZero(BossMonster bossMonster, Player player, int count) {
+		if (bossMonster.bossHpZero()) {
+			OutputView.printRaidSuccessMessage(player.getName(), count);
+			return true;
+		}
+		return false;
+	}
+
+	private int attackPlayerByBoss(Player player) {
+		int damage = randomDamageGenerator.generate();
+		player.attackedByBossMonster(damage);
+		return damage;
+	}
+
+	private void attackBossByPlayer(BossMonster bossMonster, Player player, AttackType type) {
+		if (player.validateRemainMP(type)) {
+			player.attack(type);
+			bossMonster.attackedByPlayer(type);
+		}
+	}
```

Comment:
마찬가지로 Player를 더 잘 활용할 수 있을 것 같아요~
```suggestion
    private void attackBossByPlayer(BossMonster bossMonster, Player player, AttackType type) {
        player.attack(bossMonster, type);
    }
```

Reply:
오.. 중간중간에 생각 나는 기능들을 끼워맞추다 보니 객체를 제대로 사용을 못했네요 ㅠㅠ 
좋은 지적 감사합니다!